utils/test_harness/tools/src/test_harness_metric.cpp

```diff
@@ -288,12 +314,25 @@ get_metric_group_info(ze_device_handle_t device,
       continue;
     }
 
+    concurrentMetricGroupHandles.push_back(metricGroupHandle);
     matchedGroupsInfo.emplace_back(
         metricGroupHandle, metricGroupProp.name, metricGroupProp.description,
         metricGroupProp.domain, metricGroupProp.metricCount);
   }
 
-  return matchedGroupsInfo;
+  concurrentMetricGroupHandles =
```

Comment:
what's the rationale to change existing tests to use getConcurrentMetricGroups() instead of creating a new test? I think the value of this API will be when we have platforms with multiple metrics domains (or metrics sources) and **effectively** confirm the groups can be collected concurrently (open different streamers / tracers. )

Reply:
The idea is not to test the functionality of getConcurrentMetricGroups. Currently in all streamer tests we are iterating over all metric groups. This was resulting in timeouts on Elmo runs. Hence the idea here is to somehow reduce the run-time by reducing the metric groups that the tests run on. Santhosh had suggested we can just run for 1 metric group from each source, which should be enough for all the tests. This new implementation does exactly this. It takes the first concurrent group returned by the getConcurrentMetricGroups API and only uses that for the tests. This today is expected to have 1 from OA and 1 from IP sampling.

```
  uint32_t metricGroupCountInConcurrentGroup = countPerConcurrentGroup[0];

  for (uint32_t i = 0; i < metricGroupCountInConcurrentGroup; i++) {
    concurrentMetricGroupList.push_back(metricGroupHandleList[i]);
  }
  return concurrentMetricGroupList;
```

Reply:
Perfect, makes sense. Just a few comments: 

- Leave some comments in the code to make your explanation easy to find. No need to mention Elmo, but the idea. 
- There is an env var to reduce the % of tests used. How does it impact here?